join_windows_test.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2017-2025 SUSE LLC. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package securejoin
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// Windows has very specific behaviour relating to volumes, and we can only
+// test it on Windows machines because filepath.* behaviour depends on GOOS.
+func TestHasDotDot_WindowsVolumes(t *testing.T) {
+	for _, test := range []struct {
+		testName, path string
+		expected       bool
+	}{
+		{"plain-dotdot", `C:..`, true},            // apparently legal
+		{"relative-dotdot", `C:..\foo\bar`, true}, // apparently legal
+		{"trailing-dotdot", `D:\foo\bar\..`, true},
+		{"leading-dotdot", `F:\..\foo\bar`, true},
+		{"middle-dotdot", `F:\foo\..\bar`, true},
```

Comment:
In Windows, there are many other ways to have a volume name. I remember `\\share\path\to\file`, but there's more; you can check e.g.  https://github.com/golang/go/blob/f6d17c540024418ca8a92371931b1b4fb076d2b6/src/path/filepath/path_test.go#L1553-L1614, or https://github.com/golang/go/blob/f6d17c540024418ca8a92371931b1b4fb076d2b6/src/internal/filepathlite/path_windows.go#L198-L250, or links provided in there).

Not sure how much of that you want to duplicate here; after all, we're not testing filepath.VolumeName.

Reply:
My main worry was the `C:..\foo` case, which seems like it should be valid but the [only documentation I could find](https://learn.microsoft.com/en-us/dotnet/standard/io/file-path-formats) felt somewhat suspect (what does "current directory in the `C:` drive" in the  `C:foo\bar` case even mean -- does Windows have separate cwds for each process on each drive?).

I can add a few more cases though.

Reply:
Huh, it turns out that Go's `\\.\C:..` handling is not what you would naively expect. Let me check on a windows machine if this is a Go stdlib bug or not (it is possible that `C:..\foo` is invalid anyway).

Reply:
Ah okay, I missed this:

> DOS device paths are fully qualified by definition and cannot begin with a relative directory segment (. or ..). Current directories never enter into their usage.

I'll adapt the failing tests...